packages/core/core/src/Parcel.js

```diff
@@ -92,6 +95,41 @@ export default class Parcel {
 
     this.#runPackage = this.#farm.mkhandle('runPackage');
 
+    await this.initializeWatcher();
+  }
+
+  async initializeWatcher() {
+    let projectRoot = this.#resolvedOptions.projectRoot;
+    if (this.#resolvedOptions.watch) {
+      // TODO: ideally these should all be absolute paths already set up on #resolvedOptions
+      let targetDirs = this.#resolvedOptions.targets.map(target =>
+        path.resolve(process.cwd(), target.distDir)
+      );
+      let cacheDir = path.resolve(
+        process.cwd(),
+        this.#resolvedOptions.cacheDir
+      );
+      let vcsDirs = ['.git', '.hg'].map(dir => path.join(projectRoot, dir));
+      let ignore = [cacheDir, ...targetDirs, ...vcsDirs];
+      this.#watcher = await watcher.subscribe(
+        projectRoot,
+        (err, events) => {
+          if (err) {
+            throw err;
```

Comment:
right now this will throw the error in the same tick that this callback fires, and will be uncaught. Maybe this could return something that signals the error. Observables would be really nice here...